clash-prelude/src/Clash/Sized/Internal/Index.hs

```diff
@@ -184,10 +191,28 @@ instance NFData (Index n) where
   -- NOINLINE is needed so that Clash doesn't trip on the "Index ~# Integer"
   -- coercion
 
-instance (KnownNat n, 1 <= n) => BitPack (Index n) where
-  type BitSize (Index n) = CLog 2 n
-  pack   = packXWith pack#
-  unpack = unpack#
+instance KnownNat n => BitPack (Index n) where
+  type BitSize (Index n) = BitSizeIndex n
+  pack   = case compareSNat (SNat @1) (SNat @n) of
+    SNatGT                   -> const 0
+    SNatLE | Dict <- fact @n -> packXWith pack#
+  unpack = case compareSNat (SNat @1) (SNat @n) of
+    SNatGT                   -> const undefined
+    SNatLE | Dict <- fact @n -> unpack#
+
+type family BitSizeIndex (n :: Nat) where
+  BitSizeIndex 0 = 0
+  BitSizeIndex n = CLog 2 n
+
+instance KnownNat n => KnownNat1 $(nameToSymbol ''BitSizeIndex) n where
+  natSing1 = let n = natVal (Proxy @n)
+              in SNatKn $ if n == 0
+                          then 0
+                          else fromInteger $ toInteger $ integerLog2 n
+  {-# INLINE natSing1 #-}
```

Comment:
Could we perhaps move this to a submodule `Orphans` (and import it here) such that we can eliminate the need for `-wno-orphans` on this whole module?

Reply:
Not required any more. The type family is now embedded into the `ghc-typelits-extra` plugin.